src/aposteriori/dnn/network/visualization/prediction_entropy.py

```diff
@@ -169,13 +127,12 @@ def visualize_model_entropy(
         Lists of annotated ampal assemblies.
     """
     pdb_paths = [_fetch_pdb(pdb_code) for pdb_code in pdb_codes]
-
     # Set up codec:
-    if dataset_metadata.atom_encoder == ["C", "N", "O"]:
+    if all(dataset_metadata.atom_encoder == ["C", "N", "O"]):
```

Comment:
Not sure what this all is doing here, the inner expression should evaluate to a bool:

```python
>>> [1, 2, 3] == [1, 2, 3]
True
>>> [1, 2, 3] == [1, 1, 1]
False
```
As the inner expression is a single value, all should raise an error as it's not iterable:
```python
>>> all([1, 2, 3] == [1, 2, 2])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'bool' object is not iterable
```

Reply:
It's also worth mentioning that the order is considered when comparing lists, so you might want to change these to sets, where the order is irrelevant.

Reply:
I chose `all` because `dataset_metadata.atom_encoder` is an `ndarray` meaning:
```
>>> dataset_metadata.atom_encoder == ["C", "N", "O"]
[ True  True  True]
```

Reply:
I like the idea of sets though, I'll be using them instead.

Reply:
Ah sorry, I didn't notice that.